### DISCUSSION TITLE
While running command via run-external, how to provide stdin to the running program

### DISCUSSION THREAD

**Discussion by emailstorbala** [nushell/nushell#11680] [Q&A]
Usually while running bash/ksh, when running a command, i can provide the stdin to command as like below

`echo yes | <command>`

How can I do that similarly in nushell?

**Comment by fdncred** [comment]
That works but commands have to support receiving input from the pipeline. Not all commands do.
```nushell
❯ ^echo yes | print
yes
```

**Comment by emailstorbala** [comment]
> That works but commands have to support receiving input from the pipeline. Not all commands do.
> 
> ```nushell
> ❯ ^echo yes | print
> yes
> ```

@fdncred This is not working!

![image](https://github.com/nushell/nushell/assets/12377900/0dd58eb7-b548-4774-9c5f-7c7a9da3dcd7)

**Reply by fdncred** [reply]
As I said, not all commands support pipeline input, `rm` does not.

**Reply by ineu** [reply]
You are running built-in `rm` @emailstorbala. This works: `echo Y | ^rm -i 1.txt`

**Comment by emailstorbala** [comment]
> echo Y | ^rm -i 1.txt

Yes, you are correct @ineu . Thank you!

The problem for me is that I need similar support for 'run-external' for dynamic command calls with multiple arguments. For example, I have another python program which expects multiple arguments and gather 1 input during runtime from stdin.

P.S: My original question was for run-external command only but somehow the discussion track got changed.
